src/app/stream-deck/actions.ts

```diff
@@ -0,0 +1,366 @@
+        const pickedItem = items.find((it) => !it.equipped);
+        pickedItem && (await dispatch(moveItemTo(pickedItem, vaultStore!, false)));
+        return;
+      }
+      case 'pullItem': {
+        const allItems = allItemsSelector(state);
+        const vaultStore = vaultSelector(state);
+        const item = allItems.find((it) => it.id === data.args.item);
+        if (item) {
+          if (currentStore.items.includes(item)) {
+            await dispatch(moveItemTo(item, vaultStore!, false));
+          } else {
+            await dispatch(moveItemTo(item, currentStore, false));
+          }
+        }
+      }
+    }
+  };
+}
+
+export const installFarmingObserver = _.once((dispatch) => {
+  observeStore(
+    (state) => state.farming.storeId,
+    (_, newState) => {
+      dispatch(
+        sendToStreamDeck({
+          farmingMode: Boolean(newState),
+        })
+      );
+    }
+  );
+});
+
+// collect and send data to the stream deck
+function refreshStreamDeck(): ThunkResult {
+  return async (dispatch, getState) => {
+    const refreshAction = () => {
+      const state = getState();
+      const store = currentStoreSelector(getState());
+      if (!store) {
+        return;
+      }
+      dispatch(
+        sendToStreamDeck({
+          postmaster: packager.postmaster(store),
+          maxPower: packager.maxPower(store, state),
+          vault: packager.vault(state),
+          metrics: packager.metrics(state),
+        })
+      );
+    };
+    clearInterval(refreshInterval);
+    refreshInterval = setInterval(refreshAction, 30000);
+    refreshAction();
+  };
+}
+
+// stop the websocket's connection with the local stream deck instance
+export function stopStreamDeckConnection(): ThunkResult {
+  return async (dispatch) => {
+    streamDeckWebSocket?.close();
+    clearInterval(refreshInterval);
+    dispatch(streamDeckDisconnected());
+  };
+}
+
+// start the websocket's connection with the local stream deck instance
+export function startStreamDeckConnection(): ThunkResult {
+  return async (dispatch, getState) => {
+    const initWS = () => {
+      const state = getState();
+
+      // if settings/manifest/profile are not loaded retry after 1s
+      if (
+        !state.dimApi.globalSettingsLoaded ||
+        !state.manifest.destiny2CoreSettings ||
+        !state.inventory.profileResponse?.profileProgression
+      ) {
+        setTimeout(initWS, 1000);
+        return;
+      }
+
+      // if stream deck is disabled stop and don't try to connect
+      if (!state.streamDeck.enabled) {
+        return;
+      }
+
+      installFarmingObserver(dispatch);
+
+      // close the existing websocket if connected
+      if (streamDeckWebSocket && streamDeckWebSocket.readyState !== WebSocket.CLOSED) {
+        streamDeckWebSocket.close();
+      }
+
+      // try to connect to the stream deck local instance
+      streamDeckWebSocket = new WebSocket('ws://localhost:9119', streamDeckLocal.identifier());
```

Comment:
Is it possible to use `wss://` here?

Reply:
I used your certs in my local plugin instance to expose a wss server